ts/src/schema/field.ts

```diff
@@ -98,11 +95,65 @@ export function UUIDType(options: FieldOptions): UUIDField {
   return Object.assign(result, options);
 }
 
+export interface IntegerOptions extends FieldOptions {
+  min?: number;
+  max?: number;
+}
+
 export class IntegerField extends BaseField implements Field {
   type: Type = { dbType: DBType.Int };
+  private validators: { (str: number): boolean }[] = [];
+
+  constructor(options?: IntegerOptions) {
+    super();
+    // for legacy callers
+    this.handleOptions(options || { name: "field" });
+  }
+
+  getOptions(): IntegerOptions {
+    return this.getOptions;
+  }
```

Comment:
Curious what this is for.

Reply:
it's used at https://github.com/lolopinto/ent/blob/main/ts/src/schema/string_field.test.ts#L276

Reply:
Still confused about this part though. `getOptions` never seems to actually be defined anywhere. How does this return anything?

Reply:
got it. i see what you mean. this seems like a very weird bug. not sure what happened here.

```
diff --git a/ts/src/schema/field.ts b/ts/src/schema/field.ts
index 29b35802..5066f75b 100644
--- a/ts/src/schema/field.ts
+++ b/ts/src/schema/field.ts
@@ -151,6 +151,7 @@ export class StringField extends BaseField implements Field {
   type: Type = { dbType: DBType.String };
   private validators: { (str: string): boolean }[] = [];
   private formatters: { (str: string): string }[] = [];
+  private options: StringOptions = { name: "field" };
 
   constructor(options?: StringOptions) {
     super();
@@ -159,7 +160,7 @@ export class StringField extends BaseField implements Field {
   }
 
   getOptions(): StringOptions {
-    return this.getOptions;
+    return this.options;
   }
 
   private handleOptions(options: StringOptions) {
@@ -190,6 +191,7 @@ export class StringField extends BaseField implements Field {
         noParams[k].apply(this);
       }
     }
+    this.options = options;
   }
 
   minLen(l: number): StringField {
diff --git a/ts/src/schema/string_field.test.
```

it should have been something like this

Reply:
https://github.com/lolopinto/ent/pull/590